scipy/stats/_continuous_distns.py

```diff
@@ -7943,27 +7944,112 @@ class argus_gen(rv_continuous):
     %(example)s
     """
     def _pdf(self, x, chi):
-        """
-        Return PDF of the argus function
-
-        argus.pdf(x, chi) = chi**3 / (sqrt(2*pi) * Psi(chi)) * x *
-                            sqrt(1-x**2) * exp(- 0.5 * chi**2 * (1 - x**2))
-        """
         y = 1.0 - x**2
-        return chi**3 / (_norm_pdf_C * _argus_phi(chi)) * x * np.sqrt(y) * np.exp(-chi**2 * y / 2)
+        A = chi**3 / (_norm_pdf_C * _argus_phi(chi))
+        return A * x * np.sqrt(y) * np.exp(-chi**2 * y / 2)
 
     def _cdf(self, x, chi):
-        """
-        Return CDF of the argus function
-        """
         return 1.0 - self._sf(x, chi)
 
     def _sf(self, x, chi):
-        """
-        Return survival function of the argus function
-        """
         return _argus_phi(chi * np.sqrt(1 - x**2)) / _argus_phi(chi)
 
+    def _rvs(self, chi):
+        chi = np.asarray(chi)
+        if chi.size == 1:
+            out = self._rvs_scalar(chi, self._size)
+        else:
+            shp, bc = _check_shape(chi.shape, self._size)
+            numsamples = int(np.prod(shp))
+            out = np.empty(self._size)
+            it = np.nditer([chi],
+                           flags=['multi_index'],
+                           op_flags=[['readonly']])
+            while not it.finished:
+                idx = tuple((it.multi_index[j] if not bc[j] else slice(None))
+                            for j in range(-len(self._size), 0))
+                r = self._rvs_scalar(it[0], numsamples)
+                out[idx] = r.reshape(shp)
+                it.iternext()
+
+        if self._size == ():
+            out = out[()]
+        return out
+
+    def _rvs_scalar(self, chi, numsamples=None):
+        # if chi <= 2.611:
```

Comment:
While I realize that this comment is important, it seems rather long to include here, maybe a distribution tutorial would be a better place for this? The upshot of placing it in a tutorial is that it documents as well as provides a reference for how a contributor can work to improve speed of distributions. The exact approach may differ depending on the distribution of course but this approach is somewhat generic (e.g. use knowledge about distribution and apply accept-reject principles, perhaps by splitting support of parameter(s) into sub-spaces with custom methods on each subspace.

Reply:
just to be clear if you want to go the route of a tutorial that need not be in this PR.

Reply:
It is long, but I'm not sure that it belongs in a tutorial, which I more see as educating users not developers.  If it is in a public document (book, journal, arXiv.org) that would be referenced here that would be fine.  Or remaining here with the implementation would also be useful.

Reply:
I think for a tutorial it is too technical, I mainly wanted to add some details for developers. I might also put the short note on a website at some point, it depends on how much time I find to work on it